load-testing/indexd/drs-endpoint.js

```diff
@@ -0,0 +1,55 @@
+const { check, group, sleep } = require('k6'); // eslint-disable-line import/no-unresolved
+const http = require('k6/http'); // eslint-disable-line import/no-unresolved
+const { Rate } = require('k6/metrics'); // eslint-disable-line import/no-unresolved
+
+const {
+  GUIDS_LIST,
+  GEN3_HOST,
+  ACCESS_TOKEN,
+  VIRTUAL_USERS,
+  SIGNED_URL_PROTOCOL,
+} = __ENV; // eslint-disable-line no-undef
+
+// __ENV.GUIDS_LIST should contain either a list of GUIDs from load-test-descriptor.json
+// or it should be assembled based on an indexd query (requires `indexd_record_url` to fetch DIDs)
+const guids = GUIDS_LIST.split(',');
+
+const myFailRate = new Rate('failed requests');
+
+export const options = {
+  rps: 90000,
+  stages: JSON.parse(VIRTUAL_USERS.slice(1, -1)),
+  thresholds: {
+    http_req_duration: ['avg<3000', 'p(95)<15000'],
+    'failed requests': ['rate<0.1'],
+  },
+  noConnectionReuse: true,
+};
+
+export default function () {
+  const url = `https://${GEN3_HOST}/index/ga4gh/drs/v1/objects/${guids[Math.floor(Math.random() * guids.length)]}/access/${SIGNED_URL_PROTOCOL}`;
+  const params = {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${ACCESS_TOKEN}`,
+    },
+  };
+  group('Sending DRS Endpoint request', () => {
+    group('http get', () => {
+      console.log(`Shooting requests against: ${url}`);
+      const res = http.get(url, params, { tags: { name: 'DRSEndpoint' } });
+      // console.log(`Request performed: ${new Date()}`);
+      myFailRate.add(res.status !== 200);
+      if (res.status !== 200) {
+        console.log(`Request response: ${res.status}`);
+        console.log(`Request response: ${res.body}`);
+      }
+      check(res, {
+        'is status 200': (r) => r.status === 200,
+      });
+    });
+    group('wait 0.3s between requests', () => {
```

Comment:
what happens if we don't do this? I imagine clients won't add time b/t their requests, they'll just backoff on errors

Reply:
Sorry for the late reply:
I think the rationale here (at least when we came up with the first exercises) is that we would recommend that customers running automated scripts could set up some pause between requests and also retry on the NGINX RATE LIMIT response (_http 503 - service unavailable - please try again_).

But you're right. We should remove that from the scripts to help us obtain a more accurate result.